internal/fwserver/server_deleteresource_test.go

```diff
@@ -334,6 +353,114 @@ func TestServerDeleteResource(t *testing.T) {
 				NewState: testEmptyState,
 			},
 		},
+		"response-newidentity": {
+			server: &fwserver.Server{
+				Provider: &testprovider.Provider{},
+			},
+			request: &fwserver.DeleteResourceRequest{
+				PriorState: &tfsdk.State{
+					Raw: tftypes.NewValue(testSchemaType, map[string]tftypes.Value{
+						"test_computed": tftypes.NewValue(tftypes.String, nil),
+						"test_required": tftypes.NewValue(tftypes.String, "test-priorstate-value"),
+					}),
+					Schema: testSchema,
+				},
+				IdentitySchema: testIdentitySchema,
+				ResourceSchema: testSchema,
+				Resource: &testprovider.Resource{
+					DeleteMethod: func(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+						if resp.Identity == nil || !resp.Identity.Raw.IsNull() {
+							resp.Diagnostics.AddError(
+								"Unexpected resp.Identity",
+								"expected resp.Identity to be a null object of the schema type.",
+							)
```

Comment:
Related to your comment above – I could see an ask of provider developers to be able to use identity to delete resources. At least once providers don't have to support the absence of identity support (i.e. older Terraform versions) anymore 🤔

Reply:
Yeah, I think that data should be populated regardless. Core still hasn't given us a strict definition of when the data will be what values, but it should be something at least 😆